boto/exception.py

```diff
@@ -85,6 +85,10 @@ def __init__(self, status, reason, body=None, *args):
         self._error_message = None
         self.message = ''
         self.box_usage = None
+        if len(args) > 0:
+            self.headers = args[0]
+        else:
+            self.headers = None
```

Comment:
change append init argument for headers at the end with default of None. "headers=None".

Then, lets just set self.headers to value of headers.